artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AckManager.java

```diff
@@ -139,7 +140,14 @@ public void endRetry() {
 
       // schedule a retry
       if (!sortRetries().isEmpty()) {
-         scheduledComponent.delay();
+         ActiveMQScheduledComponent scheduleComponentReference = scheduledComponent;
+         if (scheduleComponentReference != null) {
+            try {
+               scheduleComponentReference.delay();
+            } catch (RejectedExecutionException thatsOK) {
+               logger.debug("Could not schedule an execution - {}", thatsOK.getMessage(), thatsOK);
```

Comment:
```suggestion
               logger.debug("Could not schedule AckManager retry task - {}", thatsOK.getMessage(), thatsOK);
```
So that the message conveys _what_ didnt happen, otherwise I'd still need to open the code.

Reply:
this is not supposed to happen.. and still a logger.debug

Reply:
By the time this is getting an error, the server would probably be going away if this is happening due to an IO Error.